File: tests/pom/PagInicio.ts
import { Page } from '@playwright/test';
import { PaginaBase } from './PaginaBase';

export class PagInicio extends PaginaBase {
  constructor(page: Page) {
    super(page, '/index.html', 'Inicio');
  }
  async estaImagenFotoVisible() : Promise<boolean> {
    return await this.page.getByAltText('Jacinto Aisa Ibañez').isVisible();
  }
  async recuperarNombre() : Promise<string | null>   {
    return await this.page.getByRole('heading', { name: "Jacinto Aisa Ibañez" }).textContent();
  }

}

File: playwright-dotnet/PlaywrightDotnet.Tests/Pom/PagInicio.cs
using System.Threading.Tasks;
using Microsoft.Playwright;

namespace PlaywrightDotnet.Tests.Pom
{
    public class PagInicio : PaginaBase
    {
        public PagInicio(IPage page) : base(page, "/index.html", "Inicio") { }

        public async Task<bool> EstaImagenFotoVisibleAsync()
        {
            return await Page.GetByAltText("Jacinto Aisa Ibañez").IsVisibleAsync();
        }

        public async Task<string?> RecuperarNombreAsync()
        {
            return await Page.GetByRole(AriaRole.Heading, new PageGetByRoleOptions { Name = "Jacinto Aisa Ibañez" }).TextContentAsync();
        }
    }
}
